### PR TITLE
[Fix-5808][Server]  When we try to transfer data using datax between  different types of data sources, the worker will exit with ClassCastException

### LINKED ISSUE #5808: [Bug][Server]  When we try to transfer data using datax between  different types of data sources, the worker will exit with ClassCastException
**Describe the bug**

When we try to transfer data using datax between  different types of data sources, the worker will exit with ClassCastException

image

```java
[INFO] 2021-07-12 23:36:28.682  - [taskAppId=TASK-545173233664_3-3-3]:[464] - try to execute sql analysis query column name
[ERROR] 2021-07-12 23:36:28.688 org.apache.dolphinscheduler.server.worker.runner.TaskExecuteThread:[171] - task scheduler failure
java.lang.RuntimeException: java.lang.ClassCastException: org.apache.dolphinscheduler.common.datasource.clickhouse.ClickhouseConnectionParam cannot be cast to org.apache.dolphinscheduler.common.datasource.mysql.MysqlConnectionParam
	at org.apache.dolphinscheduler.common.datasource.DatasourceUtil.getConnection(DatasourceUtil.java:84)
	at org.apache.dolphinscheduler.server.worker.task.datax.DataxTask.tryExecuteSqlResolveColumnNames(DataxTask.java:557)
	at org.apache.dolphinscheduler.server.worker.task.datax.DataxTask.parsingSqlColumnNames(DataxTask.java:465)
	at org.apache.dolphinscheduler.server.worker.task.datax.DataxTask.buildDataxJobContentJson(DataxTask.java:286)
	at org.apache.dolphinscheduler.server.worker.task.datax.DataxTask.buildDataxJsonFile(DataxTask.java:215)
	at org.apache.dolphinscheduler.server.worker.task.datax.DataxTask.handle(DataxTask.java:166)
	at org.apache.dolphinscheduler.server.worker.runner.TaskExecuteThread.run(TaskExecuteThread.java:159)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run$$$capture(FutureTask.java:266)
	at java.util.concurrent.FutureTask.run(FutureTask.java)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.ClassCastException: org.apache.dolphinscheduler.common.datasource.clickhouse.ClickhouseConnectionParam cannot be cast to org.apache.dolphinscheduler.common.datasource.mysql.MysqlConnectionParam
	at org.apache.dolphinscheduler.common.datasource.mysql.MysqlDatasourceProcessor.getConnection(MysqlDatasourceProcessor.java:113)
	at org.apache.dolphinscheduler.common.datasource.DatasourceUtil.getConnection(DatasourceUtil.java:82)
	... 12 common frames omitted
```

**Which version of Dolphin Scheduler:**
latest dev
**Additional context**

**Requirement or improvement**
- Please describe about your requirements or improvement suggestions.

### PR DESCRIPTION
**Pull request by echohlne**
## Purpose of the pull request
Fix #5808
Close #5808
## Brief change log
This problem comes after #5116 , the improvement is great, but the source dataType and target dataType  was't  used correctly.

## Verify this pull request
Manually verified the change by testing locally.

### CODE REVIEW

**Comment by echohlne** [comment]
cc @ruanwenjun

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/task/datax/DataxTask.java [hunk 1]
```diff
@@ -554,7 +554,7 @@ public String loadJvmEnv(DataxParameters dataXParameters) {
         sql = sql.replace(";", "");
 
         try (
-                Connection connection = DatasourceUtil.getConnection(DbType.valueOf(dataXParameters.getDtType()), baseDataSource);
+                Connection connection = DatasourceUtil.getConnection(DbType.valueOf(dataXParameters.getDsType()), baseDataSource);
```
Comment (ruanwenjun) [inline dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/task/datax/DataxTask.java]: Can we modify the method 
```java
public String[] tryExecuteSqlResolveColumnNames(BaseConnectionParam baseDataSource, String sql)
``` 
to 
```java
public String[] tryExecuteSqlResolveColumnNames(DbType dbType, BaseConnectionParam baseDataSource, String sql)
```
Using `dataXParameters.getDsType()` here seems also confusing.

Comment (echohlne) [reply dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/task/datax/DataxTask.java]: @ruanwenjun I agree with what you said, the code has been updated.

**Review by ruanwenjun — APPROVED** [review]
LGTM

**Comment by echohlne** [comment]
ping
@CalvinKirs

**Review by CalvinKirs — APPROVED** [review]
LGTM

**Pull request merged**